thentos-core/src/Thentos/Transaction.hs

```diff
@@ -192,48 +190,19 @@ confirmUserEmailChange timeout token = do
         0 -> throwError NoSuchToken
         _ -> impossible "email change token exists multiple times"
 
-data UpdateUserFieldOp =
-    UpdateUserFieldName UserName
-  | UpdateUserFieldEmail UserEmail
-  | UpdateUserFieldPassword (HashedSecret UserPass)
-  deriving (Eq, Show)
 
--- | Update one of the attributes stored for a user.
-updateUserField :: UserId -> UpdateUserFieldOp -> ThentosQuery e ()
-updateUserField uid op = do
+-- | Change password. Should only be called once the old password has been
+-- verified.
+changePassword :: UserId -> HashedSecret UserPass -> ThentosQuery e ()
+changePassword uid newpass = do
     modified <- q
     case modified of
         1 -> return ()
         0 -> throwError NoSuchUser
-        _ -> impossible "updateUserField: unique constraint on id violated"
+        _ -> impossible "changePassword: unique constraint on id violated"
```

Comment:
What else would it be? In this branch, more than 1 records were updated, which means the uniqueness constraint on `id` was violated.